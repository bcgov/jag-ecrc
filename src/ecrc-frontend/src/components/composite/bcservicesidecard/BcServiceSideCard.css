p > button {
  display: flex;
}

.dark-button {
  background-color: #2d476f;
  border-color: #294266;
}
