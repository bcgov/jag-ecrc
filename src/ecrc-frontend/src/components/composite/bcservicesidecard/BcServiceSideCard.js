/* eslint-disable react/jsx-curly-newline */
import React from "react";

import "./BcServiceSideCard.css";
import SideCard from "../../base/sideCard/SideCard";
import { Button } from "../../base/button/Button";

export default function BcServiceSideCard() {
  const button = {
    label: "READ MORE",
    buttonStyle: "btn btn-primary dark-button",
    buttonSize: "btn btn-md",
    type: "submit"
  };

  const redirectButton = (
    <Button
      key="bc-services-card-link"
      button={button}
      onClick={() =>
        window.open(
          "https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
        )
      }
    />
  );

  const bcservice = {
    heading: "Get a BC Services Card",
    content: [
      "B.C. residents who have lived in the province for at least six months must use a BC Services Card to log in to the online qualification tool. Learn how to get a card.",
      redirectButton
    ],
    type: "blue"
  };

  return <SideCard key="bc-services-card" sideCard={bcservice} />;
}
