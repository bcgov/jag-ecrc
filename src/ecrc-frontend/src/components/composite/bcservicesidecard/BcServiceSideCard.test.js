import React from "react";
import { create } from "react-test-renderer";

import BcServiceSideCard from "./BcServiceSideCard";

describe("BcServiceSideCard Component", () => {
  test("Matches the snapshot", () => {
    const bcservicesidecard = create(<BcServiceSideCard />);
    expect(bcservicesidecard.toJSON()).toMatchSnapshot();
  });
});
