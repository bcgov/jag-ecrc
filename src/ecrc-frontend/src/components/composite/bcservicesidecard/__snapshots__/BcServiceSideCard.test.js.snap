// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BcServiceSideCard Component Matches the snapshot 1`] = `
<div
  className="dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-12"
      style={
        Object {
          "paddingTop": "30px",
          "position": "relative",
        }
      }
    >
      <p />
      <section
        className="submit-container"
        id="blue-section"
      >
        <h2
          className="heading-style"
        >
          Get a BC Services Card
        </h2>
        <div
          className="submit-content"
        >
          <p>
            B.C. residents who have lived in the province for at least six months must use a BC Services Card to log in to the online qualification tool. Learn how to get a card.
            <button
              className="btn btn-primary dark-button btn btn-md"
              onClick={[Function]}
              type="submit"
            >
              READ MORE
            </button>
          </p>
        </div>
      </section>
    </div>
  </div>
</div>
`;
